resqpy/olio/vector_utilities.py

```diff
@@ -523,13 +524,85 @@ def points_in_triangles(p, t, da, projection = 'xy', edged = False):
         return np.all(cwtd < 0.0, axis = 2)
 
 
+@njit
+def point_in_polygon(x, y, polygon):
+    """Calculates if a point in within a polygon in 2D.
+    
```

Comment:
Does this function require the polygon to be convex? (It doesn't matter for our triangles but would be good to make clear in docstring for other uses.)

Reply:
I think so but I could add some tests for that?

Reply:
Yes, I think it would be good to know, so a unit test directly calling this function for a few different shapes and points would be good.  By the way, we also have similar functionality in the olio/point_inclusion module.